File: ruby/testcode/unsafereflect_012.rb
require_relative 'shared'

# vuln-code-snippet start ruby_reflect_case_class
def select_handler(req)
  type = req.param('type')
  handler = case type # vuln-code-snippet safe-line ruby_reflect_case_class
            when 'json' then :handle_json
            when 'xml' then :handle_xml
            when 'csv' then :handle_csv
            else return BenchmarkResponse.bad_request('unknown')
            end
  BenchmarkResponse.ok("handler: #{handler}")
end
# vuln-code-snippet end ruby_reflect_case_class

File: php/testcode/unsafereflect_012.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_reflect_class_alias
function unsafereflect012(BenchmarkRequest $req): BenchmarkResponse {
    $source = $req->param('source_class');
    class_alias($source, 'AliasedHandler'); // vuln-code-snippet vuln-line php_reflect_class_alias
    $handler = new AliasedHandler();
    return BenchmarkResponse::ok($handler->run());
}
// vuln-code-snippet end php_reflect_class_alias
